Build: add database items to homepage

udenizdemirbilek opened the pull request:
closes #4 

Add items from database to homepage as cards, there are links underneath the cards to forward to
product search page/ Buyer's Product Messages Page ON PROGRESS.
Screenshot test for HomePageItems provided, App test has been changed to reflect new content.

udenizdemirbilek committed:
Build: add database items to homepage

Add items from database to homepage as cards, there are links underneath the cards to forward to
product search page/ Buyer's Product Messages Page ON PROGRESS.
Screenshot test for HomePageItems provided, App test has been changed to reflect new content.

ceyhungulbas reviewed (changes requested):
Ellerine saglik Deniz! Looks like you really worked on this. Really good job. With your comments I think functionality will work with firebase. 

I just want to suggest only one thing: A little CSS improvement -card's background color, border radius, white lines at the edge of the text- can be good.

*The pic I showed is about card's background color.

![resim](https://user-images.githubusercontent.com/44509662/128742785-8373ed2d-baf5-410c-a62d-da242b1bbbb0.png)

halitbatur commented on public/index.html:
```diff
@@ -24,7 +24,8 @@
     -->
-    <title>React App</title>
+  
+    <title>Giftify</title>
```
this should be in a seperate PR

halitbatur commented on public/manifest.json:
```diff
@@ -1,23 +1,7 @@
-      "sizes": "64x64 32x32 24x24 16x16",
-      "type": "image/x-icon"
-    },
-    {
```
same with this file this should have been done in another PR

halitbatur reviewed (changes requested):
I didn't take much look cause indentation needs to be fixed, please make sure to edit the ```yarn run fix``` script in order to check for ```scss``` extenion files instead of ```css```

udenizdemirbilek committed:
Build: add database items to homepage

udenizdemirbilek committed:
Merge github.com:ReCoded-Org/capstone-turkey-giftify into HomePageItems

udenizdemirbilek committed:
Build: add database items to homepage

udenizdemirbilek committed:
Build: add database items to homepage

udenizdemirbilek commented:
Fixed issues Please Take Another Look @halitbatur @ceyhungulbas

ceyhungulbas reviewed (approved):
Deniz first of all im so sorry cause I didnt see the new version. Sorry for delay... Ellerine saglik. I just checked and looks good to me!

halitbatur commented on src/components/HomePageItemCard/HomePageItemCard.jsx:
```diff
@@ -0,0 +1,24 @@
+function HomePageItemCard({ itemName = "", id }) {
+  return (
+    <Card className="homePageItemCard">
+      <Card.Img className="itemImage mx-auto" variant="top" src={coffeepot} />
```
I think here in the src it should be coming from the props too in case you have different items with different images, and you can also have a deafult image in case no image is passed

udenizdemirbilek replied:
You're totally right!

halitbatur commented on src/components/HomePageItems/HomePageItems.jsx:
```diff
@@ -0,0 +1,29 @@
+import HomePageItemCard from "../HomePageItemCard/HomePageItemCard";
+
+function HomePageItems() {
+  const exampleItems = [
```
This array should be coming from HomePage component via props and then used to render the items

udenizdemirbilek committed:
Build: add database items to homepage, draft request page

fix prop drilling, homepage reads items from the database at start with useEffect,
passes down to homePageItems to render homePageItemCard
homePageItemCard passes item id's to RequestItem when clicked on links

udenizdemirbilek committed:
Build: add database items to homepage, draft request page

fix prop drilling, homepage reads items from the database at start with useEffect,
passes down to homePageItems to render homePageItemCard
homePageItemCard passes item id's to RequestItem when clicked on links

udenizdemirbilek commented:
Fixed issues Please Take Another Look @halitbatur @ceyhungulbas
Also added read functions from firebase, item id is passed down as prop to RequestItemPage

udenizdemirbilek committed:
Build: add database items to homepage, draft request page

fix prop drilling, homepage reads items from the database at start with useEffect,
passes down to homePageItems to render homePageItemCard
homePageItemCard passes item id's to RequestItem with router when clicked on links

halitbatur commented on src/App.js:
```diff
@@ -1,25 +1,24 @@
+      <Container fluid>
+        <Switch>
+          <Route exact path="/" component={HomePage} />
+          <Route path="/request-item/:id">
```
lets use exact path here in order to make sure we dont create a page called `/request-item/5/<anything>`

halitbatur commented on src/components/HomePageItems/HomePageItems.jsx:
```diff
@@ -0,0 +1,18 @@
+function HomePageItems({ dbItems }) {
+  return (
+    <Row xs={2} md={3} xl={4} className="homePageItems">
+      {dbItems.map((e) => (
```
please lets use a better variable name instead of `e`

udenizdemirbilek replied:
used dbItem

halitbatur commented on src/containers/HomePage/HomePage.jsx:
```diff
@@ -0,0 +1,36 @@
+import db from "../../firebase";
+
+// After deciding on the database structure,the example array should be deleted and
+//  replaced with a function that reads from firebase, takes 8 items by random
```
extra space here

udenizdemirbilek replied:
wow you have the eyes of a hawk! :)

halitbatur commented on src/containers/HomePage/HomePage.jsx:
```diff
@@ -0,0 +1,36 @@
+  const [dbItems, setDbItems] = useState([]);
+
+  useEffect(() => {
+    db.collection("homePageItems")
```
lets use async await instead of .then()

udenizdemirbilek committed:
draft push for inspection

udenizdemirbilek committed:
Build: database item cards for homepage

fetch items from firestore database and display them as cards on homepage
cards are linked to request item pages with their id's

udenizdemirbilek committed:
Merge branch 'dev' into HomePageItems

udenizdemirbilek commented:
fixed latest issues, 
fetching from firebase uses async await now
PTAL

ceyhungulbas commented:
Deniz I just check it and LGTM. Functionality is working, styling is good, folder structure is delicious! 👌🏼 Ellerine saglik abi.

ceyhungulbas reviewed (approved):
Deniz I just check it and LGTM. Functionality is working, styling is good, folder structure is delicious! 👌🏼 Ellerine saglik abi.

busesisi reviewed (commented):
The css and functionality looks very good Deniz. I see some tests are failing but tother than that, ellerine sağlık :)

udenizdemirbilek committed:
Build: add database items to homepage

fetch items from firestore database and display them as cards on homepage
cards are linked to request item pages with their id's

udenizdemirbilek commented:
fixed latest issues,
fetching from firebase uses async await now
PTAL @halitbatur

halitbatur commented:
@udenizdemirbilek 
![image](https://user-images.githubusercontent.com/61846570/129891583-327de8cb-6b81-4b98-85f6-9d528742a858.png)
I guess ur new set up is dealing with end of line differently than wsl and that causes this bug where u have a lot of lines extra in the changed lines, please take a look here : https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings to fix the issue

udenizdemirbilek committed:
fix line endings

udenizdemirbilek committed:
fix line endings2

udenizdemirbilek committed:
Normalize all the line endings

udenizdemirbilek committed:
Merge branch 'dev' into HomePageItems

udenizdemirbilek committed:
update snapshot

udenizdemirbilek committed:
Merge branch 'HomePageItems' of https://github.com/ReCoded-Org/capstone-turkey-giftify into HomePageItems

udenizdemirbilek committed:
fix App

udenizdemirbilek committed:
fix App2

udenizdemirbilek commented:
@halitbatur PTAL line endings are fixed, fingers are crossed

halitbatur reviewed (approved):
good work !

udenizdemirbilek commented:
> good work !

This is a small step for mankind but a big step for me :D
